src/libmongoc/tests/test-mongoc-client.c

```diff
@@ -2254,7 +2254,7 @@ test_mongoc_client_mismatched_me (void)
    mock_server_run (server);
    uri = mongoc_uri_copy (mock_server_get_uri (server));
    mongoc_uri_set_option_as_utf8 (uri, "replicaSet", "rs");
-   client = test_framework_client_new_from_uri (uri, NULL);
+   client = mongoc_client_new_from_uri (uri);
```

Comment:
Why this change?

Reply:
This is because even if no server API is specified to test_framework_client_new_from_uri() (i.e.the second parameter is NULL), it still picks up the MONGOC_SERVER_API environment variable. Unfortunately, there are MANY tests only check for legacy hello and fail or abort() if an OP_MSG hello is sent instead. This function, and its char* and pool conterparts, doesn't check for that variable, so we get legacy hello-- as expected.

Reply:
Okay, these have been revised. Thank you both for the suggestion.